models/fivetran_log__audit_table.sql

```diff
@@ -7,13 +19,31 @@ with sync_log as (
     from {{ ref('stg_fivetran_log__log') }}
 
     where event_subtype in ('sync_start', 'sync_end', 'write_to_table_start', 'write_to_table_end', 'records_modified')
+
+    {% if is_incremental() %}
+
+    -- Capture the latest timestamp in a call statement instead of a subquery for optimizing BQ costs on incremental runs
+    {%- call statement('max_sync_start', fetch_result=True) -%}
+        select date(max(sync_start)) from {{ this }}
+    {%- endcall -%}
+
+    {%- set query = load_result('max_sync_start') -%}
+
+    -- store it as a singular value
+    {%- set max_sync_start = query['data'][0][0] -%}
+
+        -- compare the new batch of data to the latest sync already stored in this model
+        and date(created_at) >= '{{ max_sync_start }}'
+
+    {% endif %}
```

Comment:
Only comment here would be to update the names of the variables to be a bit more clear. See my suggestions below:
```suggestion
    {%- call statement('max_sync_start', fetch_result=True) -%}
        select date(max(sync_start)) from {{ this }}
    {%- endcall -%}

    -- load the result from the above query into a new variable
    {%- set query_result = load_result('max_sync_start') -%}

    -- the query_result is stored as a dataframe. Therefore, we want to now store it as a singular value.
    {%- set max_sync_start = query_result['data'][0][0] -%}

        -- compare the new batch of data to the latest sync already stored in this model
        and date(created_at) >= '{{ max_sync_start }}'

    {% endif %}
```